Sources/Chat/Chat.swift

```diff
@@ -0,0 +1,65 @@
+
+import Foundation
+import WalletConnectUtils
+import WalletConnectKMS
+import WalletConnectRelay
+import Combine
+
+
+class Chat {
+    private var publishers = [AnyCancellable]()
+    let registry: Registry
+    let engine: Engine
+    let kms: KeyManagementService
+    var onInvite: ((Invite)->())?
+    var onNewThread: ((Thread)->())?
+    var onConnected: (()->())?
+
+    init(registry: Registry,
+         relayClient: RelayClient,
+         kms: KeyManagementService,
+         logger: ConsoleLogging = ConsoleLogger(loggingLevel: .off),
+         keyValueStorage: KeyValueStorage) {
+        self.registry = registry
+        
+        self.kms = kms
+        let serialiser = Serializer(kms: kms)
+        let networkingInteractor = NetworkingInteractor(relayClient: relayClient, serializer: serialiser)
+        let topicToInvitationPubKeyStore = KeyValueStore<String>(defaults: keyValueStorage, identifier: StorageDomainIdentifiers.topicToInvitationPubKey.rawValue)
+        let inviteStore = KeyValueStore<Invite>(defaults: keyValueStorage, identifier: StorageDomainIdentifiers.invite.rawValue)
+        self.engine = Engine(registry: registry,
+                             networkingInteractor: networkingInteractor,
+                             kms: kms,
+                             logger: logger,
+                             topicToInvitationPubKeyStore: topicToInvitationPubKeyStore,
+                             inviteStore: inviteStore)
+        relayClient.socketConnectionStatusPublisher.sink { [unowned self] status in
```

Comment:
Could we move subscriptions from `init()`? For example in `subscriptions()` method

Reply:
of course